lib/compress/zstd_compress.c

```diff
@@ -2305,6 +2307,11 @@ static size_t ZSTD_compressBlock_internal(ZSTD_CCtx* zc,
             zc->bmi2);
 
     if (frame &&
+        /* We don't want to emit our first block as a RLE even if it qualifies because
+         * doing so will cause the decoder to throw a "should consume all input error."
```

Comment:
minor: mentions that it's only for the __CLI__ decoder.
This is a very big difference compared to the `libzstd` decoder.
This makes it more probable that we may remove this restriction in some (distant) future.